src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2101,6 +2101,10 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 dest_type = ASR::down_cast<ASR::ttype_t>(
                         ASR::make_Character_t(al, loc, right_type2->m_kind,
                         dest_len, nullptr));
+            } else if ((left_is_int && ASRUtils::expr_value(left) == nullptr) 
+                || (right_is_int && ASRUtils::expr_value(right) == nullptr)) {
```

Comment:
Would simply doing the following work?

```suggestion
            } else if (left_is_int || right_is_int) {
```

Reply:
Thanks for suggesting this simplification @Shaikh-Ubaid ! I realized that as we are checking for compile-time integers above, we might as well have only `else` here. It works.